plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergMetadata.java

```diff
@@ -0,0 +1,564 @@
+import org.apache.iceberg.TableScan;
+import org.apache.iceberg.Transaction;
+import org.apache.iceberg.UpdateLocation;
+import org.apache.iceberg.UpdatePartitionSpec;
+import org.apache.iceberg.UpdateProperties;
+import org.apache.iceberg.UpdateSchema;
+import org.apache.iceberg.encryption.EncryptionManager;
+import org.apache.iceberg.io.FileIO;
+import org.apache.iceberg.io.LocationProvider;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static io.airlift.json.JsonCodec.jsonCodec;
+import static io.trino.plugin.iceberg.TrinoHiveCatalog.DEPENDS_ON_TABLES;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThatThrownBy;
+
+public class TestIcebergMetadata
+{
+    private static final CatalogSchemaTableName TESTING_MATERIALIZED_VIEW_NAME = new CatalogSchemaTableName(
+            "test_catalog_name",
+            "test_schema_name",
+            "test_table_name");
+    private static final ConnectorMaterializedViewDefinition TEST_MATERIALIZED_VIEW_DEFINITION = new ConnectorMaterializedViewDefinition(
+            "originalSql",
+            Optional.of(TESTING_MATERIALIZED_VIEW_NAME),
+            Optional.empty(),
+            Optional.empty(),
+            ImmutableList.of(new Column("test_column", TypeId.of("double"))),
+            Optional.empty(),
+            "",
+            ImmutableMap.of());
+
+    private IcebergMetadata testingIcebergMetadata;
+    private TestingTrinoCatalog testingTrinoCatalog;
+
+    @BeforeClass
+    public void setup()
+    {
+        testingTrinoCatalog = new TestingTrinoCatalog();
+        testingIcebergMetadata = new IcebergMetadata(new TestingTypeManager(), jsonCodec(CommitTaskData.class), testingTrinoCatalog);
+    }
+
+    @Test
+    public void testGetMaterializedViewFreshness()
+    {
+        testingTrinoCatalog.returnWhenGetMaterializedView(Optional.of(TEST_MATERIALIZED_VIEW_DEFINITION));
+
+        assertThatThrownBy(() ->
+                testingIcebergMetadata.getMaterializedViewFreshness(
+                        TestingConnectorSession.SESSION,
+                        TESTING_MATERIALIZED_VIEW_NAME.getSchemaTableName()))
+                .isInstanceOf(MaterializedViewNotFoundException.class)
+                .hasMessage(format("Materialized View '%s' not found", TESTING_MATERIALIZED_VIEW_NAME.getSchemaTableName()));
+
+        testingTrinoCatalog.returnWhenGetMaterializedView(Optional.empty());
+
+        assertThat(
+                testingIcebergMetadata.getMaterializedViewFreshness(
+                        TestingConnectorSession.SESSION,
+                        TESTING_MATERIALIZED_VIEW_NAME.getSchemaTableName()))
+                .isEqualTo(new MaterializedViewFreshness(false));
```

Comment:
shouldn't exception be throw here?

Reply:
No since `getMaterializedViewFreshness` -> `getMaterializedViewToken` -> `getMaterializedView` will return empty, not `getTableHandle`;

It would be nice to stub method `getMaterializedViewToken` so the test flow is easier to understand, however it's in the same class, so it would be tricky and require reflection.